labelling-wc-apps/README.md

```diff
@@ -0,0 +1,66 @@
+# Labelling workload cluster's apps
+
+The purpose of this RFC is to propose a labelling scheme to safely identify a workload cluster's defining apps.
+
+This is required for:
+
+- deleting a workload cluster on CAPI;
+- displaying a workload cluster's defining apps in a client like the web UI (happa).
+
+## Deleting a workload cluster on CAPI
+
+To delete a workload cluster on CAPI, we need to identify the corresponding cluster app.
+
+### Current state
+
+On CAPI we are relying on the resource names (in the org namespace) for cluster app selection which is not recommended. There is no reference between cluster CR and cluster app CR - unlike other cluster's app resources, cluster app CR doesn't have `giantswarm.io/cluster` label. It cannot be set at the moment, since it will break cluster deletion.
+
+### Proposal
+
+Require the `giantswarm.io/cluster` label to be present in all App resources related to a workload cluster, including the cluster app.
```

Comment:
Just to add some context, I believe that we don't add the label to the cluster-app because `cluster-apps-operator` adds a finalizer to the Cluster CR once it starts reconciling a cluster, which is removed from the Cluster CR once all the apps labeled `giantswarm.io/cluster` are deleted. But the cluster-app won't be removed until the Cluster CR is removed. And that won't happen until the finalizer is removed. And that won't happen until the cluster-app is removed. So we have a chicken - egg kind of problem.

How do you propose to rewrite the deletion logic so that it works after adding the `giantswarm.io/cluster` label?

Reply:
I think we can be more explicit about not deleting the cluster-app. At the moment it kinda happens as a side effect, it's assumed that you have a knowledge that the cluster-app doesn't have a `giantswarm.io/cluster` label. Maybe instead of selecting the apps labeled with `giantswarm.io/cluster` label, we can select the apps labeled with `giantswarm.io/cluster` and that don't have `application.giantswarm.io/cluster-app` label which I suggest to add.